Add Pop CLI to Polkadot Docs – Quickstart Development Section (#538)
polkadot-developers/polkadot-docs: develop/toolkit/parachains/quickstart/pop-cli.md

---
title: Quickstart Parachain Development with Pop CLI
description: Quickly bootstrap parachain projects, scaffold templates, deploy local networks, and streamline development workflows using Pop CLI.
---

# Quickstart Parachain Development with Pop CLI

## Introduction

> Pop CLI is a powerful command-line tool designed explicitly for rapid parachain development within the Polkadot ecosystem. It addresses essential developer needs by providing streamlined commands to set up development environments, scaffold parachain templates, and manage local blockchain networks.

Author: Nice! Included a link

Pop CLI simplifies parachain development with features like:

- Quick initialization of parachain development environments

- Project scaffolding from predefined parachain templates

- Easy deployment and management of local development networks

Developers can quickly begin coding and testing, significantly reducing setup overhead.

> ### Install Pop CLI

Author: Solved in https://github.com/polkadot-developers/polkadot-docs/pull/538/commits/770862091934e615ec9bc1b997cc1b4f406f941b

> To install Pop CLI run the following command

Author: Solved in https://github.com/polkadot-developers/polkadot-docs/pull/538/commits/770862091934e615ec9bc1b997cc1b4f406f941b

```bash
cargo install --force --locked pop-cli
```

> Confirm that Pop CLI is installed by running `pop --help` in your terminal

Author: Solved in https://github.com/polkadot-developers/polkadot-docs/pull/538/commits/770862091934e615ec9bc1b997cc1b4f406f941b

```bash
pop --help
```

> ### Set up your development environment

Author: Fixed here in https://github.com/polkadot-developers/polkadot-docs/pull/538/commits/9f1f9d36cffe27b4f9fc52184e4d3ea16e76daf1 and reviewed the rest of new titles are consistent.

> To develop and build Polkadot SDK-based it's essential to prepare your local environment with the necessary tools and dependencies. The [Install Polkadot SDK Dependencies](https://docs.polkadot.com/develop/parachains/install-polkadot-sdk/) guide walks you through this setup step-by-step.

Author: Solved in https://github.com/polkadot-developers/polkadot-docs/pull/538/commits/770862091934e615ec9bc1b997cc1b4f406f941b

However, you can automate this entire process by running:
```bash
pop install
```

> This command provides an interactive experience that checks for all necessary dependencies and installs them for you. It’s the fastest and easiest way to get your development environment ready for building parachains with Pop CLI.

Author: Solved in https://github.com/polkadot-developers/polkadot-docs/pull/538/commits/770862091934e615ec9bc1b997cc1b4f406f941b

--8<-- 'code/develop/toolkit/parachains/quickstart/pop-cli/install.html'

> 

Author: Solved in https://github.com/polkadot-developers/polkadot-docs/pull/538/commits/770862091934e615ec9bc1b997cc1b4f406f941b

### Initialize a Project

Start a new project quickly using Pop CLI `pop new parachain` command:

> --8<-- 'code/develop/toolkit/parachains/quickstart/pop-cli/new-parachain.html'

Author: Nice suggestion. Included an explanation in https://github.com/polkadot-developers/polkadot-docs/pull/538/commits/9f1f9d36cffe27b4f9fc52184e4d3ea16e76daf1

> Once the project is generated, move into the new directory and build your parachain:

Author: Solved in https://github.com/polkadot-developers/polkadot-docs/pull/538/commits/770862091934e615ec9bc1b997cc1b4f406f941b

```
cd my-parachain

> pop build --release

Author: Nice suggestion. Included an explanation in https://github.com/polkadot-developers/polkadot-docs/pull/538/commits/9f1f9d36cffe27b4f9fc52184e4d3ea16e76daf1

```

> Pop CLI integrates the [Zombienet-SDK](https://github.com/paritytech/zombienet-sdk) allowing you to easily launch ephemeral local networks for development and testing. To start a network, simply run:

Author: Solved in https://github.com/polkadot-developers/polkadot-docs/pull/538/commits/770862091934e615ec9bc1b997cc1b4f406f941b

description: Quickly bootstrap parachain projects, scaffold templates, deploy local networks, and streamline development workflows using Pop CLI.
---

# Quickstart Parachain Development With Pop CLI

## Introduction

[Pop CLI](https://onpop.io/cli/){target=\_blank} is a powerful command-line tool designed explicitly for rapid parachain development within the Polkadot ecosystem. It addresses essential developer needs by providing streamlined commands to set up development environments, scaffold parachain templates, and manage local blockchain networks.

Pop CLI simplifies parachain development with features like:

- Quick initialization of parachain development environments

- Project scaffolding from predefined parachain templates

- Easy deployment and management of local development networks

Developers can quickly begin coding and testing, significantly reducing setup overhead.

### Install Pop CLI

To install Pop CLI, run the following command:

```bash
cargo install --force --locked pop-cli
```

Confirm that Pop CLI is installed by running `pop --help` in your terminal:

```bash
pop --help
```

### Set Up Your Development Environment

> To develop and build Polkadot SDK-based it's essential to prepare your local environment with the necessary tools and dependencies. The [Install Polkadot SDK Dependencies](/develop/parachains/install-polkadot-sdk/){target=\_blank} guide walks you through this setup step-by-step.

Reviewer: ```suggestion
To develop and build Polkadot SDK-based chains, preparing your local environment with the necessary tools and dependencies is essential. The [Install Polkadot SDK Dependencies](/develop/parachains/install-polkadot-sdk/){target=\_blank} guide walks you through this setup step-by-step.
```

Author: Fixed in https://github.com/polkadot-developers/polkadot-docs/pull/538/commits/cf6091a8a444f08447db2bdaa88438254130b286

However, you can automate this entire process by running:

```bash
pop install
```

> This command provides an interactive experience that checks for all necessary dependencies and installs them for you. It’s the fastest and easiest way to get your development environment ready for building parachains with Pop CLI.

Reviewer: ```suggestion
This command provides an interactive experience that checks and installs all necessary dependencies for you. It’s the fastest and easiest way to prepare your development environment for building parachains with Pop CLI.
```

Author: Fixed in https://github.com/polkadot-developers/polkadot-docs/pull/538/commits/cf6091a8a444f08447db2bdaa88438254130b286

--8<-- 'code/develop/toolkit/parachains/quickstart/pop-cli/install.html'

### Initialize a Project

> Start a new project quickly using Pop CLI `pop new parachain` command:

Reviewer: ```suggestion
Start a new project quickly using Pop CLI's `pop new parachain` command:
```

Author: Fixed in https://github.com/polkadot-developers/polkadot-docs/pull/538/commits/cf6091a8a444f08447db2bdaa88438254130b286

--8<-- 'code/develop/toolkit/parachains/quickstart/pop-cli/new-parachain.html'

> The command above scaffolds a new parachain project using the default template included with Pop CLI. For more specialized implementations, additional templates are available, you can explore them by running `pop new parachain --help`.

Reviewer: ```suggestion
The command above scaffolds a new parachain project using the default template included with Pop CLI. For more specialized implementations, additional templates are available; you can explore them by running `pop new parachain --help`.
```

Author: Fixed in https://github.com/polkadot-developers/polkadot-docs/pull/538/commits/cf6091a8a444f08447db2bdaa88438254130b286

Once the project is generated, move into the new directory and build your parachain:

```
cd my-parachain
pop build --release
```

!!! note

>     Under the hood, `pop build --release` runs `cargo build --release`, but `pop build` adds functionality specific to Polkadot SDK projects, such as deterministic runtime builds and automatic management of feature flags like `benchmark` or `try-runtime`.

Reviewer 2: We do have a page discussing deterministic runtimes - https://docs.polkadot.com/develop/parachains/deployment/build-deterministic-runtime/. @AlexD10S, if you think adding this link to the paragraph would be helpful, feel free to do so. If not, you can just ignore it

```suggestion
    Under the hood, `pop build --release` runs `cargo build --release`, but `pop build` adds functionality specific to Polkadot SDK projects, such as [deterministic runtime builds](/develop/parachains/deployment/build-deterministic-runtime/){target=\_blank} and automatic management of feature flags like `benchmark` or `try-runtime`.
```

> Pop CLI integrates the [Zombienet-SDK](https://github.com/paritytech/zombienet-sdk){target=\_blank} allowing you to easily launch ephemeral local networks for development and testing. To start a network, simply run the following:

Reviewer: ```suggestion
Pop CLI integrates the [Zombienet SDK](https://github.com/paritytech/zombienet-sdk){target=\_blank}, allowing you to easily launch ephemeral local networks for development and testing. To start a network, simply run the following:
```

Author: Fixed in https://github.com/polkadot-developers/polkadot-docs/pull/538/commits/cf6091a8a444f08447db2bdaa88438254130b286

```bash
pop up network -f ./network.toml
```

This command will automatically fetch the necessary binaries and spin up a Polkadot network with your configured parachains.

> You can also interact with your local network using Pop CLI `pop call chain` command:

Reviewer: ```suggestion
You can also interact with your local network using Pop CLI's `pop call chain` command:
```

Author: Fixed in https://github.com/polkadot-developers/polkadot-docs/pull/538/commits/cf6091a8a444f08447db2bdaa88438254130b286
